### ISSUE TITLE
Migration to Guice 7+

### ISSUE THREAD

**Issue by Nava2** [cashapp/misk#2863]
Wanted to make an issue for tracking migration to Guice 7.0.0. 

@ThePumpingLemma migrated to using Guice 6.0.0 (#2846), however, there remains issues with projects wanting to use `misk-inject` with Guice 7+. The largest issue is that when a project tries to use Guice 7+, it can not inject classes that are labelled with `javax.inject.XXX` annotations. Guice 6+ happily accepts _both_ `jakarta` and `javax`.

## Proposed path forwards

It _seems_ like the path to migration would be twp steps:

1. Replace all usages of `javax.inject.XXX` with `com.google.inject.XXX`
    * This enables projects to transition to using `jakarta.inject` in their own projects, while consuming `misk-inject` @ 6.0.0
2. _Eventually_ cut over to Guice 7.0.0 at _some point_ 
3. _Optional:_ Replace `com.google.inject.XXX` with `jakarta.inject.XXX`

## Open question

1.  misk doesn't seem to use something like `detekt` to enable [forbidding imports](https://detekt.dev/docs/rules/style/#forbiddenimport), is there a way for us to enforce only using `com.google.inject.XXX`? 
  * IntelliJ can also be adjusted to ignore certain imports for "Auto import" ([Link](https://www.jetbrains.com/help/idea/creating-and-optimizing-imports.html#exclude-from-auto-import)), but there is no intellij configuration within misk by default

P.S. I'm happy to do the find-and-replace 😄

**Comment by r3mariano** [comment]
👋 I'm one of the engineers working on frameworks at Cash. 

We'd have to migrate our internal downstream services before we're able to do the migration to Guice 7. Will talk to the team about where to prioritise that. Given Guice 7 doesn't really have anything that isn't in Guice 6 at the moment, I'd lean towards doing the cutover later.

Using detekt would be great to check for imports!

**Comment by r3mariano** [comment]
I'm curious btw, why `com.google.inject` instead of `jakarta.inject` if they both work? I think I'd prefer to use the more generic one of the two where possible.

**Comment by Nava2** [comment]
If we used Jakarta, it would force usage of 6.0+.. which seems fine... 🤦🏻 Admittedly, I'm biased as our downstream services are not on 6.0 yet and it would force our update - which we need to do.

**Comment by Nava2** [comment]
I agree strongly with holding off on the cutover until later, but doing the initial migration to Jakarta seems good to get done.
